Examples/Detectors/ContextualDetector/include/ActsExamples/ContextualDetector/AlignmentDecorator.hpp

```diff
@@ -50,42 +55,86 @@ class AlignmentDecorator : public IContextDecorator {
+//       double tz = cfg.gSigmaZ != 0 ? cfg.gSigmaZ * gauss(rng) : 0.;
+//       // Add a translation - if there is any
+//       if (tx != 0. or ty != 0. or tz != 0.) {
+//         const auto& tMatrix = trf.matrix();
+//         auto colX = tMatrix.block<3, 1>(0, 0).transpose();
+//         auto colY = tMatrix.block<3, 1>(0, 1).transpose();
+//         auto colZ = tMatrix.block<3, 1>(0, 2).transpose();
+//         Acts::Vector3 newCenter = tMatrix.block<3, 1>(0, 3).transpose() +
+//                                   tx * colX + ty * colY + tz * colZ;
+//         trf.translation() = newCenter;
+//       }
+//       // now modify it - rotation around local X
+//       if (cfg.aSigmaX != 0.) {
+//         trf *=
+//             Acts::AngleAxis3(cfg.aSigmaX * gauss(rng), Acts::Vector3::UnitX());
+//       }
+//       if (cfg.aSigmaY != 0.) {
+//         trf *=
+//             Acts::AngleAxis3(cfg.aSigmaY * gauss(rng), Acts::Vector3::UnitY());
+//       }
+//       if (cfg.aSigmaZ != 0.) {
+//         trf *=
+//             Acts::AngleAxis3(cfg.aSigmaZ * gauss(rng), Acts::Vector3::UnitZ());
+//       }
+//     }
+//   }
+// };
+// }  // namespace Contextual
+// }  // namespace ActsExamples
+
+static void applyTransform(Acts::Transform3& trf, const Config& cfg,
                              RandomEngine& rng, unsigned int iov) {
-    std::normal_distribution<double> gauss(0., 1.);
-    if (iov != 0 or not cfg.firstIovNominal) {
-      // the shifts in x, y, z
-      double tx = cfg.gSigmaX != 0 ? cfg.gSigmaX * gauss(rng) : 0.;
-      double ty = cfg.gSigmaY != 0 ? cfg.gSigmaY * gauss(rng) : 0.;
-      double tz = cfg.gSigmaZ != 0 ? cfg.gSigmaZ * gauss(rng) : 0.;
-      // Add a translation - if there is any
-      if (tx != 0. or ty != 0. or tz != 0.) {
-        const auto& tMatrix = trf.matrix();
-        auto colX = tMatrix.block<3, 1>(0, 0).transpose();
-        auto colY = tMatrix.block<3, 1>(0, 1).transpose();
-        auto colZ = tMatrix.block<3, 1>(0, 2).transpose();
-        Acts::Vector3 newCenter = tMatrix.block<3, 1>(0, 3).transpose() +
-                                  tx * colX + ty * colY + tz * colZ;
-        trf.translation() = newCenter;
-      }
-      // now modify it - rotation around local X
-      if (cfg.aSigmaX != 0.) {
-        trf *=
-            Acts::AngleAxis3(cfg.aSigmaX * gauss(rng), Acts::Vector3::UnitX());
-      }
-      if (cfg.aSigmaY != 0.) {
-        trf *=
-            Acts::AngleAxis3(cfg.aSigmaY * gauss(rng), Acts::Vector3::UnitY());
-      }
-      if (cfg.aSigmaZ != 0.) {
-        trf *=
-            Acts::AngleAxis3(cfg.aSigmaZ * gauss(rng), Acts::Vector3::UnitZ());
-      }
+    if (iov < cfg.misalignmentInstructions.size()) {
+      const MisalignmentStruct& misalignment = cfg.misalignmentInstructions[iov];
+
+      // place where we take care of transformations before misalignment (for each event)
+      cfg.preMisalignmentTransforms[iov] = trf;
```

Comment:
Is this only for groups then? For the sensors we can get this directly from the surfaces?